### PR TITLE
chore: TypeScript 4.2.3

### PR DESCRIPTION
**Pull request by agg23**
Upgrades all Cypress packages and examples to use TypeScript 4.2.3 (latest at time of writing)

The `@cypress/vue` package (`npm/vue`) contains a minor hack to force tslib to the latest version, due to larger issues with the monorepo's dependency resolution.

### CODE REVIEW

File: packages/network/lib/agent.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export const isResponseStatusCode200 = (head: string) => {
 }
 
 export const regenerateRequestHead = (req: http.ClientRequest) => {
-  delete req._header
+  req._header = {}
```
Comment (elevatebart) [inline packages/network/lib/agent.ts:94]: Is this part of the typescript migration?

**Review by elevatebart — COMMENTED** [review]
I really appreciate you doing this upgrade. We really need it.
What is the reason for getting rid of the `delete` statements? Is it for the TS migration?

**Comment by agg23** [comment]
Yes. `delete` is not allowed unless the type is optional, not `null`able.

**Review by elevatebart — DISMISSED** [review]
Then it looks good when CI passes

**Review by flotwig — DISMISSED** [review]
LGTM once tests pass

File: packages/network/lib/uri.ts [hunk 2]
```diff
@@ -46,7 +46,10 @@ export function removePort (urlObject) {
 
   // set host to undefined else url.format(...) will ignore the port property
   // https://nodejs.org/api/url.html#url_url_format_urlobject
+  // Additionally, the types are incorrect (don't include undefined), so we add TS exceptions
+  /* @ts-ignore */
```
Comment (lmiller1990) [inline packages/network/lib/uri.ts:50]: Is this a case where we can use `ts-expect-error`?

**Review by lmiller1990 — COMMENTED** [review]
Great. What feature are you wanting to use from TS 4.2?

File: packages/network/lib/uri.ts (hunk 2 above)
Comment (agg23) [reply packages/network/lib/uri.ts:50]: We could, but it feels like an odd scenario since you're fighting an incorrect type.

**Review by lmiller1990 — DISMISSED** [review]

**Review by elevatebart — APPROVED** [review]

**Pull request merged**